### PR TITLE
lakectl local: handle signals

### LINKED ISSUE #6356: lakectl local: Handle signals
Create a mechanism to handle signals in lakectl local commands and present a meaningful message

[Edited by @arielshaqed to talk about "signals"]

### PR DESCRIPTION
**Pull request by johnnyaug**
Resolves #6356.
For now, print an error message.

### CODE REVIEW

**Review by N-o-Z — CHANGES_REQUESTED** [review]
@johnnyaug Thanks,
Can we put it in a general function that runs the logic of checking the ctx Done channel in the background and prints the error if needed?

Thinking of something in the likes of:


```
func localInterruptListener(ctx context.Context) context.Context {
	cCtx, stop := signal.NotifyContext(ctx, os.Interrupt, syscall.SIGTERM)
	go func() {
		defer stop()
		<-cCtx.Done()
		if cCtx.Err() != nil {
			DieFmt("message: %w", cCtx.Err())
		}
	}()
	return cCtx
}
```

File: cmd/lakectl/cmd/local_pull.go [hunk 1]
```diff
@@ -63,8 +70,11 @@ var localPullCmd = &cobra.Command{
 			}
 			return nil
 		})
-		s := local.NewSyncManager(cmd.Context(), client, syncFlags.parallelism, syncFlags.presign)
+		s := local.NewSyncManager(ctx, client, syncFlags.parallelism, syncFlags.presign)
```
Comment (N-o-Z) [inline cmd/lakectl/cmd/local_pull.go:73]: Consider passing the sigCtx only to the SyncManager and not for other operations

Comment (johnnyaug) [reply cmd/lakectl/cmd/local_pull.go:73]: I considered that, but in such case - an interruption will not be handled during the preliminary requests.

Comment (N-o-Z) [reply cmd/lakectl/cmd/local_pull.go:73]: If you put the New right before the Sync call - you could potentially use this context less

Comment (N-o-Z) [reply cmd/lakectl/cmd/local_pull.go:73]: Also if you use the suggested method with the function - we will call Stop() immediately after receiving the first signal and revert the signal override

**Review by arielshaqed — DISMISSED** [review]
I don't understand: how does this help the user?  Usually we handle these signals in order to perform necessary cleanups: for instance to leave the terminal in a usable state (`more`, `less`, `vim`), or to leave local state in a usable state (`git`, `vim`), or similar.  Here this is not required.  Presumably if I interrupt the program I should know that it was interrupted.  And in any case my shell will let me know what happened.

Here AFAICT we need no cleanup.  Suppose `lakectl local pull` is stopped:
* Humans get a _less informative_ error message than the shell would give.  And they need to learn a new way in which programs can fail.
* Programs get a _less informative_ exit status of `lakectl` than the `wait(2)` would give.

By contrast, this is what `git` does if I interrupt it.  I am running it through `gh`, so that's another level of sugaring that still allows me to see the interrupt:
```sh
❯ gh repo clone treeverse/lakefs
Cloning into 'lakeFS'...
remote: Enumerating objects: 69014, done.
remote: Counting objects: 100% (3592/3592), done.
^Cfetch-pack: unexpected disconnect while reading sideband packet
```
OTOH, despite keeping the clean shell experience, `git` _does_ clean up its mess.

**Comment by johnnyaug** [comment]
@arielshaqed I see your point, and I'm absolutely in favor of less sugaring and being more transparent about what happened. I don't understand however what you mean by that:
> And in any case my shell will let me know what happened.

Before this change, we would get strange behaviors when the program was interrupted. For example, we could get "0 objects downloaded successfully" (even though it was interrupted in the middle of downloading objects). So _some_ handling is clearly needed.

Let me ask it this way, what do you expect to see when the program is interrupted?

**Comment by johnnyaug** [comment]
EDIT: I think @arielshaqed is right: before the PR, it shows a message of `signal: interrupt`, which I think is good enough. I think we can close this PR. We can strive to also clean up our mess, but not sure that's for now.

**Review by N-o-Z — APPROVED** [review]
Awesome!

File: cmd/lakectl/cmd/local_commit.go [hunk 2]
```diff
@@ -105,7 +105,8 @@ var localCommitCmd = &cobra.Command{
 				c <- change
 			}
 		}()
-		s := local.NewSyncManager(cmd.Context(), client, syncFlags.parallelism, syncFlags.presign)
+		sigCtx := localHandleSyncInterrupt(cmd.Context())
```
Comment (arielshaqed) [inline cmd/lakectl/cmd/local_commit.go:108]: Won't this print an incorrect and misleading error message?

**Review by arielshaqed — COMMENTED** [review]
Frankly, I dislike this PR: it trades correctness and functionality for a more polite error message.  I am removing my request for changes, as I understand that my objection is to the requested feature.

**Pull request merged**